index.js

```diff
@@ -1,15 +1,28 @@
-require('./lib/array_fill');
-require('./lib/array_intersect');
-require('./lib/array_qsort');
-require('./lib/array_unique');
-require('./lib/fs_scandir');
-require('./lib/object_watch');
-require('./lib/string_format');
-require('./lib/string_hashcode');
-require('./lib/string_pipe');
-
-module.exports = {
-  flatten : require('./lib/flatten'),
-  get     : require('./lib/get_deep'),
-  set     : require('./lib/set_deep')
+module.exports = function (opts) {
+  opts = ( 'object' === typeof opts ) ? ( opts || {} ) : {};
+
+  require('./lib/array_fill');
+  require('./lib/array_intersect');
+  require('./lib/array_qsort');
+  require('./lib/array_unique');
+  require('./lib/fs_scandir');
+
+  require('./lib/string_format');
+  require('./lib/string_hashcode');
+
+  // skip prototype pollution of methods used by browserify
+  if (!opts.build) {
+    require('./lib/string_pipe');
+  }
+  if (!opts.test) {
+    require('./lib/object_watch');
```

Comment:
I'm missing the exact reason why this is can by disabled. Maybe add a comment what crashed during the build you mentioned?

Reply:
While running tests on the api-client package if not omitting object.watch.


````sh
milan@freeside ~/w/R/trackthis-api-client> npm test

> trackthis-api-client@0.4.0 test /home/milan/workspace/Ratus/trackthis-api-client
> nyc node_modules/.bin/serial-mocha test/* --sort | tee test-output.txt && ./node_modules/.bin/nyc report --reporter=clover && ! grep -q -oP "\d+\sfailing" test-output.txt && rm test-output.txt || { echo "Test output is located in 'test-output.txt'" ; exit 1 ; }

/home/milan/workspace/Ratus/trackthis-api-client/node_modules/mocha/lib/utils.js:59
exports.watch = function (files, fn) {
              ^

TypeError: Cannot assign to read only property 'watch' of object '#<Object>'
    at Object.<anonymous> (/home/milan/workspace/Ratus/trackthis-api-client/node_modules/mocha/lib/utils.js:59:15)
    at Module._compile (module.js:652:30)
    at Module.replacementCompile (/home/milan/workspace/Ratus/trackthis-api-client/node_modules/nyc/node_modules/append-transform/index.js:58:13)
    at module.exports (/home/milan/workspace/Ratus/trackthis-api-client/node_modules/nyc/node_modules/default-require-extensions/js.js:8:9)
    at Object.<anonymous> (/home/milan/workspace/Ratus/trackthis-api-client/node_modules/nyc/node_modules/append-transform/index.js:62:4)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/milan/workspace/Ratus/trackthis-api-client/node_modules/mocha/lib/reporters/base.js:10:13)
    at Module._compile (module.js:652:30)
    at Module.replacementCompile (/home/milan/workspace/Ratus/trackthis-api-client/node_modules/nyc/node_modules/append-transform/index.js:58:13)
    at module.exports (/home/milan/workspace/Ratus/trackthis-api-client/node_modules/nyc/node_modules/default-require-extensions/js.js:8:9)
    at Object.<anonymous> (/home/milan/workspace/Ratus/trackthis-api-client/node_modules/nyc/node_modules/append-transform/index.js:62:4)
    at Module.load (module.js:565:32)
````

Now actually inspecting the function a little better ( :) ) I noticed the

````
 * Edited by finwo:
 *   Added a handler response check, to not break the value non-returning calls
 *   Made the code pass JSHINT linting
 *   Allow it to be overwritten in debug mode, for tests like mocha & nyc
 */

(function (overwritable) {
````
So I guess I can pass this through opts somehow, and see if that works.